docs/guide/units_tutorial.rst

```diff
@@ -0,0 +1,232 @@
+    rydberg \> 2.179846-18 m2-kg/sec2\\
+    pi \> 3.141593\\
+    lambert \> 3.183099+3 candela/m2
+
+and it is more convenient to define *constant* constants in the UNITS block rather than in the CONSTANTS block --- there is less chance of a typo, and they do not appear in SCoP where they can be inadvertently changed. For example:
+
+.. code::
+    python
+
+    UNITS {
+        F      = (faraday) (coulomb)
+        PI     = (pi) (1)
+        e      = (e) (coulomb)
+        R      = (k-mole) (joule/degC)
+        C      = (c) (cm/sec)
+    }
+
+Here, ``C`` is the speed of light in cm/sec and ``R`` is the Gas constant.
+
+Constant factors are defined in the UNITS block in the following manner.
+
+.. code::
+    python
+
+    UNITS {
+        F   = 96520    (coul)
+        PI  = 3.14159  ()
+        foot2inch = 12 (inch/foot)
+    }
+
+Note that one could also write the last example as
+
+.. code::
+    python
+
+    foot2inch = (1) (inch/foot)
+
+This shows that it can sometimes take too much clear thinking to specify dimensionless conversion factors. To avoid misunderstanding, conversion factors ( dimensionless factors used to convert between conformable units) can most clearly be written
+
+.. code::
+    python
+
+    UNITS {
+        foot2inch = (foot) -> (inch)
+    }
+
+With the above, a statement such as 
+
+.. code::
+    python
+
+    i = 5*foot2inch*f
+
+makes sense. ie. if f=2 then i should end up as 120 and there is no complaint by the units checker.
+
+Function and Argument units
+===================
+
+Standard mathematical functions are dimensionless and take dimensionless arguments. In those special circumstances (such as printf()) where dimensions don't matter or where there is some overreaching reason why units should not be checked, one can turn off all unit checking with the statement, ``UNITSOFF``. Unit checking is turned back on with UNITSON. An example of this last case are Hodgkin-Huxley rates which involve terms like ``exp(-v/18)``. In the model description the argument has the dimensions of mV and it may seem like too much trouble to give all the numbers explicit units as in ``exp(-v/18(mV))``.
+
+User declared functions and arguments can be given units with the syntax
+
+.. code::
+    python
+
+    FUNCTION f(a1 (u1), a2 (u2), ...) (uf) { statements }
+    PROCEDURE p(a1 (u1), a2 (u2), ...) { statements }
+
+In every instance, the absence of units implies that the argument or function is dimensionless.
+
+Units for KINETIC blocks
+===================
+
+First, it is ascertained that the quantity units are the same for each reactant. Quantity units are the reactant units times that reactant's COMPARTMENT size. Then the flux units are determined (quantity units / independent units) and it is ascertained that the rate units are consistent.
+
+Units in NEURON models
+===================
+
+If a NEURON block exists in the model description the following variables are checked for compatible units
+
+.. code::
+    python
```

Comment:
There's actually a units table in the documentation... we should link there instead?